test/compatibility/test_fast_truncate_info.py

```diff
@@ -0,0 +1,174 @@
+        '''
+
+        # Checkout the relevant branches.
+        older_branch = 'mongodb-6.0'
+        newer_branch = 'mongodb-7.0'
+        self.prepare_wt_branch(older_branch, self.cmake_args)
+        self.prepare_wt_branch(newer_branch, self.cmake_args)
+
+        # Run the first part of the test on the 6.0 branch. This generates database files
+        # that do not contain fast-truncate information.
+        self.run_method_on_branch(older_branch, 'generate_older_database_files')
+
+        # Run the second part of the test on the 7.0 branch. This reads 6.0 database files
+        # and appends fast-truncate information onto database records when needed.
+        self.run_method_on_branch(newer_branch, 'write_fast_truncate_info')
+
+        # Run the third part of the test on the 6.0 branch. This attempts to read database
+        # files containing fast-truncate information from an older version that does not
+        # support writing out fast-truncate information. We should be able to read the
+        # information though we will simply ignore it.
+        self.run_method_on_branch(older_branch, 'read_fast_truncate_info_from_older')
+
+    def generate_older_database_files(self):
+        import wiredtiger
+
+        conn = wiredtiger.wiredtiger_open('.', 'create,')
+        session = conn.open_session()
+
+        print(f'Running on {wiredtiger.wiredtiger_version()[0]}')
+        session.create(self.uri, self.config + ',key_format=i,value_format=S')
+
+        c = session.open_cursor(self.uri)
+        for i in range(1, self.nentries):
+            c[i] = 'i' + str(i)
+        c.close()
+
+        session.checkpoint()
+
+        session.close()
+        conn.close
+
+    def write_fast_truncate_info(self):
+        import wiredtiger
+        from wiredtiger import stat
+
+        # Open the connection on the database directory containing the database
+        # generated by the older binary.
+        new_conn = wiredtiger.wiredtiger_open('.', 'create=false,statistics=(all)')
+
+        print(f'Running on {wiredtiger.wiredtiger_version()[0]}')
+
+        new_conn.set_timestamp('oldest_timestamp='+ self.timestamp_str(200))
+        new_conn.set_timestamp('stable_timestamp='+ self.timestamp_str(200))
+
+        session = new_conn.open_session()
+        session.begin_transaction()
+        
+        # Truncate a big range of rows.
+        start = session.open_cursor(self.uri, None)
+        start.set_key(10)
+        end = session.open_cursor(self.uri, None)
+        end.set_key(self.nentries - 10)
+        session.truncate(None, start, end, None)
+        start.close()
+        end.close()
+
+        session.commit_transaction('commit_timestamp=' + self.timestamp_str(250))
+
+        # Ensure that fast-truncate happened.
+        stat_cursor = session.open_cursor('statistics:', None, None)
+        fastdelete_pages = stat_cursor[stat.conn.rec_page_delete_fast][2]
+        assert(fastdelete_pages > 0)
```

Comment:
You should use functions such as  `self.assertGreater` that we use in the tests instead of Python's own `assert`.